cranelift/codegen/src/egg.rs

```diff
@@ -134,9 +132,21 @@ impl<'a> FuncEGraph<'a> {
                 };
                 let id = self.egraph.add(node, &mut self.node_ctx);
 
-                if side_effect {
-                    self.side_effect_ids.push(id);
-                }
+                let id = match (side_effect, id) {
+                    (true, id) => {
+                        let id = id.get();
+                        self.side_effect_ids.push(id);
+                        id
+                    }
+                    (false, NewOrExisting::New(id)) => {
+                        // Apply all optimization rules immediately; the
+                        // aegraph (acyclic egraph) works best when we do
```

Comment:
how about "e-dag" instead of "acyclic e-graph" ?

Reply:
That could work too but I have some more acronymming based on aegraph (writeup to come) so I want to stick with this for now :-)